### ISSUE TITLE
This extension must be completely revised.

### ISSUE THREAD

**Issue by stefsava** [Kosai106/vscode-ruby-syntax-replacer#4]
The example in README.md denote a poor knowledge of ruby language.

```ruby
# Old syntax
data = [
  { 1 => "en", 2 => "to", 3 => "tre" },
  { :one => "un", :two => "deux", :three => "trois" },
  { "weights" => ["pound", "kilogram"], "lengths" => ["meter", "mile"] },
  { :de => { :one => "eins", :two => "zwei", :three => "drei" } }
]

# New syntax
data = [
  { 1: "en", 2: "to", 3: "tre" }, # syntax error: symbols can not start with numerical digits 
  { one: "un", two: "deux", three: "trois" },
  { "weights": ["pound", "kilogram"], "lengths": ["meter", "mile"] }, # bad replacement: keys was strings, but replaced with symbols
  { de: { one: "eins", two: "zwei", three: "drei" } }
]
```

**Comment by Kosai106** [comment]
You're right on the part about symbols not being able to start with a number, that was an oversight on my part. However I'd say it's a bit extreme to demand that everything should be revised based off of that.

As for your second point, if doing this would cause issues with your code, _(Doesn't in mine)_ then just don't run the command on the affected lines. Simple.

**Comment by patbl** [comment]
It's surprising that this extension subtly alters the behavior of hashes that have string keys. This could cause people who think that the altered code is semantically equivalent to introduce bugs into their code. I think a more conservative approach—applying the transformation only to keys that are symbols—would lead to less wailing and gnashing of teeth.

That's how [this Emacs package](https://github.com/purcell/ruby-hash-syntax) behaves.

**Comment by Kosai106** [comment]
@patbl Would this be a more suitable change for you? https://github.com/Kosai106/vscode-ruby-syntax-replacer/pull/11

The result would now look like this:
```rb
data = [
  { 1: "en", 2: "to", 3: "tre" },
  { one: "un", two: "deux", three: "trois" },
  { "weights" => ["pound", "kilogram"], "lengths" => ["meter", "mile"] },
  { de: { one: "eins", two: "zwei", three: "drei" } }
]
```

**Comment by chabgood** [comment]
> You're right on the part about symbols not being able to start with a number, that was an oversight on my part. However I'd say it's a bit extreme to demand that everything should be revised based off of that.
> 
> As for your second point, if doing this would cause issues with your code, _(Doesn't in mine)_ then just don't run the command on the affected lines. Simple.

When you say run the command, what command is that?

**Comment by Kosai106** [comment]
> When you say run the command, what command is that?

The `replace ruby syntax` command you get when installing the plugin.
If you have a line highlighted, it will only replace the syntax on that line.

<img width="629" alt="Screenshot 2021-12-14 at 10 32 38" src="https://user-images.githubusercontent.com/6379824/145971775-5789ac22-4651-429b-a9a4-af5122697796.png">
